File_Q&A.py

```diff
@@ -0,0 +1,42 @@
+import streamlit as st
+import openai
+import json
+from components.Sidebar import sidebar
+
+api_key = sidebar()
+
+st.title("📝 File Q&A with OpenRouter")
+uploaded_file = st.file_uploader("Upload an article", type="txt")
+with st.form("chat_input", clear_on_submit=True):
+    a, b = st.columns([4, 1])
+    question = a.text_input(
+        "Ask something about the article",
+        placeholder="Can you give me a short summary?",
+        disabled=not uploaded_file,
+    )
+    b.form_submit_button("Ask", use_container_width=True)
+
+if uploaded_file and question and not api_key:
+    st.info("Please add your OpenRouter API key to continue.")
+
+if uploaded_file and question and api_key:
+    article = uploaded_file.read().decode()
+    openai.api_base = "https://openrouter.ai/api/v1"
```

Comment:
let's move this to a constant. Let's also move all the openai calls to a new file, to prevent code duplication between these pages